worker/lease/manager.go

```diff
@@ -140,6 +141,13 @@ func (manager *Manager) Wait() error {
 
 // loop runs until the manager is stopped.
 func (manager *Manager) loop() error {
+	if collector, ok := manager.config.Store.(prometheus.Collector); ok {
+		// The store implements the collector interface, but the lease.Store
+		// doen't expose those.
+		manager.config.PrometheusRegisterer.Register(collector)
```

Comment:
Yeah, it's definitely nicer to do this here - for some reason it didn't occur to me that the type collecting the metrics didn't need to be the one to register itself.